From ab5dd371b9310e44013979286922cc44bfd64c03 Mon Sep 17 00:00:00 2001
From: Zhang Yanmin <yanmin.zhang@intel.com>
Date: Fri, 15 Apr 2016 03:27:54 +0800
Subject: [PATCH 1/4] Sched: dont ouput cpu sched info by default

When system memory is used up, usersapce might dump system info
by sysrq. One info is of cpu sched, which is not very useful.
When there are lots of running processes, such cpu sched dumping
spends lots of time and make the system worse.

The patch fixes it by dumping cpu sched info only when
sched_debug_enabled is true.

Change-Id: Iaef825af294b2f919b3963a917f46fa0170541d4
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-29317
Signed-off-by: Zhang Yanmin <yanmin.zhang@intel.com>
Reviewed-on: https://android.intel.com:443/495036
---
 kernel/sched/core.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/kernel/sched/core.c b/kernel/sched/core.c
index 43932db..87993c5 100644
--- a/kernel/sched/core.c
+++ b/kernel/sched/core.c
@@ -4443,6 +4443,10 @@ void sched_show_task(struct task_struct *p)
 	show_stack(p, NULL);
 }
 
+#ifdef CONFIG_SCHED_DEBUG
+static __read_mostly int sched_debug_enabled;
+#endif
+
 void show_state_filter(unsigned long state_filter)
 {
 	struct task_struct *g, *p;
@@ -4468,7 +4472,8 @@ void show_state_filter(unsigned long state_filter)
 	touch_all_softlockup_watchdogs();
 
 #ifdef CONFIG_SCHED_DEBUG
-	sysrq_sched_debug_show();
+	if (sched_debug_enabled)
+		sysrq_sched_debug_show();
 #endif
 	rcu_read_unlock();
 	/*
@@ -5152,8 +5157,6 @@ static cpumask_var_t sched_domains_tmpmask; /* sched_domains_mutex */
 
 #ifdef CONFIG_SCHED_DEBUG
 
-static __read_mostly int sched_debug_enabled;
-
 static int __init sched_debug_setup(char *str)
 {
 	sched_debug_enabled = 1;
-- 
1.9.1

